extensions/sql-migration/src/dialog/validationResults/validateIrDialog.ts

```diff
@@ -363,11 +364,10 @@ export class ValidateIrDialog {
 	}
 
 	private async _validateDatabaseMigration(): Promise<void> {
-		const sqlConnections = await azdata.connection.getConnections();
-		const currentConnection = sqlConnections.find(
-			value => value.connectionId === this._model.sourceConnectionId);
+		const currentConnection = await getSourceConnectionProfile();
 		const sourceServerName = currentConnection?.serverName!;
-		const trustServerCertificate = currentConnection?.options?.trustServerCertificate === true;
+		const encryptConnection = currentConnection?.options?.encrypt === true || currentConnection?.options?.encrypt === 'true';
+		const trustServerCertificate = currentConnection?.options?.trustServerCertificate === true || currentConnection?.options?.trustServerCertificate === 'true';
```

Comment:
nit, maybe create helper methods to reading each of these?

Reply:
good call